src/explorer/CosmosDBTreeItem.ts

```diff
@@ -159,10 +171,48 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
         return result;
     }
 
+    private detectPostgresConnections(appSettings: { [propertyName: string]: string }): IDetectedConnection[] {
+        const connectionStringPrefix = 'postgres://';
+        const portDefault = '5432';
+
+        const result: IDetectedConnection[] = [];
+        const regexp = new RegExp(`(.+)${this._pgHostSuffix}`, 'i');
+        for (const key of Object.keys(appSettings)) {
+            const match = key && key.match(regexp);
+            if (match) {
+                const prefix = match[1];
+                const hostKey = prefix + this._pgHostSuffix;
+                const dbNameKey = prefix + this._pgDbNameSuffix;
+                const userKey = prefix + this._pgUserSuffix;
+                const passKey = prefix + this._pgPassSuffix;
+                const portKey = prefix + this._pgPortSuffix;
+                const host: string | undefined = appSettings[hostKey];
+                const dbName: string | undefined = appSettings[dbNameKey];
+                const username: string | undefined = appSettings[userKey];
+                const password: string | undefined = appSettings[passKey];
+                const port: string | undefined = appSettings[portKey];
+
+                if (host) {
+                    const keys: string[] = [hostKey];
+                    keys.push(userKey, passKey, portKey, dbNameKey);
```

Comment:
I think this was better how you had it before. We only need to add to keys if the value is defined